ntoskrnl/se/access.c

```diff
@@ -23,10 +23,10 @@
  * @param[in] Process
  * Valid process object where subject context is to be captured.
  *
- * @param[in,out] AccessState
+ * @param[out] AccessState
  * An initialized returned parameter to an access state.
  *
- * @param[in] AuxData
+ * @param[out] AuxData
```

Comment:
`_Out_` implies that the routine is fully responsible to initialize and return the contents to the caller on that parameter, whereas the function actually expects a valid auxiliary data to be given to the access state as done by `AccessState->AuxData = AuxData;` and the used privileges being assigned to that auxiliary data. So I would argue using `_Inout_` is more fitting for this case? 🤷

Reply:
Note: in terms of doxygen comments, this would translate into `@param[in,out]`

Reply:
I've thought about this a bit, but I think `_Out_` really only means "contents not valid before the call, contents valid after the call". And the function completely initializes the structure and doesn't expect any of it to be valid on input.
The `__drv_aliasesMem` expresses the "this pointer is copied to elsewhere", so I think the combination of the two expresses the right thing.